replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/Worker.java

```diff
@@ -0,0 +1,377 @@
+        LOGGER.debug("Failed to close adapter", ie);
+      }
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Set this workers running flag to false if it is not currently transferring an item
+   *
+   * @return {@code true} if the worker was cancelled, otherwise {@code false}
+   */
+  boolean cancelIfProcessing() {
+    if (processing) {
+      this.running = false;
+      return true;
+    }
+    return false;
+  }
+
+  private boolean supported(Task task) {
+    if (OperationType.UNKNOWN.equals(task.getOperation())) {
+      return false;
+    }
+
+    List<MetadataInfo> infos = task.metadatas().collect(Collectors.toList());
+    if (infos.size() > 1) {
+      LOGGER.warn("More than one metadata is currently not supported by this worker");
+      return false;
+    }
+
+    DDFMetadataInfo ddfInfo =
+        infos.stream()
+            .filter(DDFMetadataInfo.class::isInstance)
+            .map(DDFMetadataInfo.class::cast)
+            .findFirst()
+            .orElse(null);
+
+    if (ddfInfo == null) {
+      LOGGER.info("Unable to handle any of the metadatas received in this task");
+      return false;
+    }
+
+    return true;
+  }
+
+  private void handle(Task task) throws InterruptedException {
+    NodeAdapter destination = getDestinationAdapter(task);
+    if (destination == null) {
+      task.unlock();
+      return;
+    }
+
+    if (!adaptersAvailable(task, destination)) {
+      return;
+    }
+
+    List<DDFMetadataInfo> infos =
+        task.metadatas()
+            .filter(DDFMetadataInfo.class::isInstance)
+            .map(DDFMetadataInfo.class::cast)
+            .collect(Collectors.toList());
+
+    final DDFMetadataInfo info = infos.get(0);
+    final Metadata metadata =
+        new MetadataImpl(
+            ((Map) info.getData()).get("raw-metadata"),
+            info.getDataClass(),
+            task.getId(),
+            Date.from(info.getLastModified()));
+
+    final OperationType type = task.getOperation();
+    switch (type) {
+      case HARVEST:
+      case REPLICATE_FROM:
+        doCreate(task, metadata, destination, local);
+        break;
+      case REPLICATE_TO:
+        doCreate(task, metadata, local, destination);
+        break;
+      case DELETE:
+        doDelete(task, metadata, destination);
+        break;
+      default:
+        // should never hit this since deserialization will return UNKNOWN
+        task.unlock();
+    }
+  }
+
+  @Nullable
+  private NodeAdapter getDestinationAdapter(Task task) {
+    Site site;
+    final String siteId = task.getQueue().getSite();
+    try {
+      site = siteManager.get(siteId);
+    } catch (NotFoundException e) {
+      LOGGER.debug("Unable to find site {} for queue", siteId, e);
+      return null;
+    } catch (ReplicationPersistenceException e) {
+      LOGGER.debug("Failed to fetch site {} for queue", siteId, e);
+      return null;
+    }
+
+    return nodeAdapters.factoryFor(site.getType()).create(site.getUrl());
```

Comment:
❗️ `getAdapterFor()` can throw `IllegalArgumentException` if it is unable to find a factory for the provided site.

Reply:
Caught it and AdapterException and unlocked task if either happens.